utilities/tools/module-indexes/module-index.tests.ps1

```diff
@@ -0,0 +1,747 @@
+    [ValidateNotNullOrEmpty()]
+    [Alias('Path')]
+    [string]$CsvFilePath
+)
+
+BeforeAll {
+    $csvContent = Import-Csv -Path $CsvFilePath
+    $rawFile = Get-Content -Path $CsvFilePath
+    $csvHeaders = $csvContent[0].PSObject.Properties.Name
+
+    $issues = & gh issue list --limit 99999 --state all --json title,body | ConvertFrom-Json
+
+    $singularExceptions = @(
+        'redis',
+        'address',
+        'publicipaddress'
+    )
+    $rawFile | Out-Null
+    $csvHeaders | Out-Null
+    $issues | Out-Null
+    $singularExceptions | Out-Null
+}
+
+Describe "Tests for the $(Split-Path $CsvFilePath -Leaf) file" {
+
+    Context 'CSV file' {
+
+        It 'CSV file must not be empty' {
+            $csvContent | Should -Not -BeNullOrEmpty
+        }
+
+        if ($CsvFilePath -match 'ResourceModules') {
+            It 'Should contain exactly the columns required for resource modules' {
+                $requiredColumns = @(
+                    'ProviderNamespace', 'ResourceType', 'ModuleDisplayName', 'ModuleName', 'ModuleStatus',
+                    'RepoURL', 'PublicRegistryReference', 'TelemetryIdPrefix', 'PrimaryModuleOwnerGHHandle',
+                    'PrimaryModuleOwnerDisplayName', 'SecondaryModuleOwnerGHHandle', 'SecondaryModuleOwnerDisplayName',
+                    'ModuleOwnersGHTeam', 'ModuleContributorsGHTeam', 'Description', 'Comments', 'FirstPublishedIn'
+                )
+                $csvHeaders | Should -Be $requiredColumns
+            }
+        } elseif ($CsvFilePath -match 'PatternModules') {
+            It 'Should contain exactly the columns required for pattern modules' {
+                $requiredColumns = @(
+                    'ModuleDisplayName', 'ModuleName', 'ModuleStatus',
+                    'RepoURL', 'PublicRegistryReference', 'TelemetryIdPrefix', 'PrimaryModuleOwnerGHHandle',
+                    'PrimaryModuleOwnerDisplayName', 'SecondaryModuleOwnerGHHandle', 'SecondaryModuleOwnerDisplayName',
+                    'ModuleOwnersGHTeam', 'ModuleContributorsGHTeam', 'Description', 'Comments', 'FirstPublishedIn'
+                )
+                $csvHeaders | Should -Be $requiredColumns
+            }
+        } elseif ($CsvFilePath -match 'UtilityModules') {
+            It 'Should contain exactly the columns required for pattern modules' {
+                $requiredColumns = @(
+                    'ModuleDisplayName', 'ModuleName', 'ModuleStatus',
+                    'RepoURL', 'PublicRegistryReference', 'TelemetryIdPrefix', 'PrimaryModuleOwnerGHHandle',
+                    'PrimaryModuleOwnerDisplayName', 'SecondaryModuleOwnerGHHandle', 'SecondaryModuleOwnerDisplayName',
+                    'ModuleOwnersGHTeam', 'ModuleContributorsGHTeam', 'Description', 'Comments', 'FirstPublishedIn'
+                )
+                $csvHeaders | Should -Be $requiredColumns
+            }
+        }
+
```

Comment:
Why? Because of the headers?

Reply:
Technically this could be under the same condition under ptn modules, but kept it separate not only for readability, but for future proofing it (there's a good chance that this will fork and will have different columns.)

Reply:
I may made a mistake and commented one line to high. I'm targeting line 67:
```pwsh
        It 'Should have at least 1 record' {
            $csvContent.Length | Should -BeGreaterOrEqual 1
        }
```